packages/core/src/agent/Dispatcher.ts

```diff
@@ -24,10 +24,14 @@ class Dispatcher {
   private eventEmitter: EventEmitter
   private logger: Logger
 
-  public constructor(messageSender: MessageSender, eventEmitter: EventEmitter, agentConfig: AgentConfig) {
+  public constructor(
+    messageSender: MessageSender,
+    eventEmitter: EventEmitter,
+    @inject(InjectionSymbols.Logger) logger: Logger
+  ) {
     this.messageSender = messageSender
     this.eventEmitter = eventEmitter
-    this.logger = agentConfig.logger
+    this.logger = logger
```

Comment:
I am a bit confused by this. Maybe I am missing something about the context, but could we not get the logger from `agentContext.logger`? 

Checking a bit through this PR it seems a bit "messy" on how we deal with the logger instance. We get it via the injection but it is also in the `agentConfig`. I am not too knowledgeable on dependency injection with tsyringe, but is this really the best way to deal with the logger instance?

Also, do we want each agent to have their instance of a logger or do we want to use a `RootContainer` (if I understand your design correctly) logger instance?

Reply:
Yeah that's something I also wasn't sure about, so good that you're calling it out. 

It's in the agent config because we configure it in the agent init config. However my thought was that the logger would be shared between the different tenants. I'm not sure if it would work well if we have a lot of logger instances, and there's ways to different context of the different agents in logs.

As mentioned in the design document we should probably look at a way to separate agent config (global, used by all tenants) and tenant config (specific to that tenant).

What would be your preferred approach here?

Reply:
Yeah I think a global configuration, maybe a Provider/manager agent aswell, and tenant specific configuration would be a good solution.

Quickly looking through the `AgentConfig` I do not see a lot that should be globally defined. That being said, I still think it has value for future configuration.